pkg/controller/utils.go

```diff
@@ -0,0 +1,79 @@
+package controller
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/deis/builder/pkg/conf"
+)
+
+const (
+	hostEnvName = "DEIS_WORKFLOW_SERVICE_HOST"
+	portEnvName = "DEIS_WORKFLOW_SERVICE_PORT"
+)
+
+// UserInfo represent the required information from a user to make a push and interact with deis/workflow
+type UserInfo struct {
+	Username    string
+	Key         string
+	FingerPrint string
+	Apps        []string
+}
+
+func controllerURLStr(additionalPath ...string) (string, error) {
+	host := os.Getenv(hostEnvName)
+	port := os.Getenv(portEnvName)
+
+	if host == "" {
+		return "", fmt.Errorf("missing required '%v' environment variable", hostEnvName)
+	}
+
+	if port == "" {
+		return "", fmt.Errorf("missing required '%v' environment variable", portEnvName)
+	}
+
+	return fmt.Sprintf("http://%s:%s/%s", host, port, strings.Join(additionalPath, "/")), nil
+}
+
+func UserInfoFromKey(key string) (*UserInfo, error) {
+	keyB64 := base64.RawURLEncoding.EncodeToString([]byte(key))
+	url, err := controllerURLStr("v2", "hooks", "key", keyB64)
+	if err != nil {
```

Comment:
Yeah it's `key`, I dropped the plural so the old hooks worked still under `keys` (the one we will remove). It also made sense since we were querying on a singular key instead of getting many keys back.